decidim-core/app/models/decidim/scope.rb

```diff
@@ -60,19 +60,19 @@ def forbid_cycles
     end
 
     def create_part_of
-      self[:part_of] = calculate_part_of
+      recalculate_part_of
       save if changed?
     end
 
     def update_part_of
-      self[:part_of] = calculate_part_of
+      recalculate_part_of
     end
 
-    def calculate_part_of
+    def recalculate_part_of
```

Comment:
I followed your steps and I have come to the same conclusion as you. :smile:
I'd just rename `recalculate_part_of` to `build_part_of` and simply ignore the fact that the `part_of` is cleared before updating it. Do you agree?